File: Tecnicatura/Javascript/clase_09/Cliente.js
class Cliente extends Persona {
    // Atributo estatico
    static contadorClientes = 0;

    // Constructor
    constructor(nombre, apellido, edad, ficharegistro) {
        super(nombre, apellido, edad);
        this._idCliente = ++Cliente.contadorClientes;
        this._fechaRegistro = this._fechaRegistro;
    }

    get idCliente() {
        return this._idCliente;
    } 

    get fechaRegistro() {
        return this._fechaRegistro;
    }

    set fechaRegistro(fechaRegistro) {
        this._fechaRegistro = fechaRegistro
    }

    toString() {
        return `
        ${super.toString()} 
        ${this._idCliente} 
        ${this._fechaRegistro}`;
    }
}    

File: Tecnicatura/Java/Clase9/Herencia/src/domain/Cliente.java
package domain;

import java.util.Date;

public class Cliente extends Persona {
    //Atributos
    private int idCliente;
    private Date fechaRegistro; //importar la clase Date
    private boolean vip; 
    private static int contadorCliente; //tipo estatico
    
    //Constructor
    public Cliente(Date fechaRegistro, boolean vip, String nombre,
            char genero, int edad, String direccion) {
        super(nombre, genero, edad, direccion);
        this.idCliente = ++Cliente.contadorCliente;
        this.fechaRegistro = fechaRegistro;
        this.vip = vip;
    }

    public int getIdCliente() {
        return this.idCliente;
    }

    public Date getFechaRegistro() {
        return this.fechaRegistro;
    }

    public void setFechaRegistro(Date fechaRegistro) {
        this.fechaRegistro = fechaRegistro;
    }

    public boolean isVip() {
        return this.vip;
    }

    public void setVip(boolean vip) {
        this.vip = vip;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Cliente{idCliente = ").append(idCliente);
        sb.append(", fechaRegistro=").append(fechaRegistro);
        sb.append(", vip=").append(vip);
        sb.append(",  ").append(super.toString());
        sb.append('}');
        return sb.toString();
    }
    
    
}
